src/fs/driver/nfs/nfs.c

```diff
@@ -551,10 +551,23 @@ static int nfs_create_dir_entry(node_t *parent_node) {
 			point += sizeof(vf);
 			predesc = (readdir_desc_t *) point;
 
-			if (predesc->file_attr.type == 1) {
-				predesc->file_attr.mode |= S_IFREG;
-			} else {
-				predesc->file_attr.mode |= S_IFDIR;
+			switch (predesc->file_attr.type) {
+				case 1:
+					/* regular file */
+					predesc->file_attr.mode |= S_IFREG;
+					break;
+				case 2:
+					/* directory */
+					predesc->file_attr.mode |= S_IFDIR;
+					break;
+				case 3:
+					/* directory */
+					predesc->file_attr.mode |= S_IFDIR;
+					break;
+				default:
+					/* unknown file type */
```

Comment:
Can you please add `log_error("Unknown file type");` here?

Reply:
added